src/mapping/RadialBasisFctMapping.hpp

```diff
@@ -245,69 +238,321 @@ void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::map(
   int polyparams = 1 + getDimensions() - deadDimensions;
 
   if (getConstraint() == CONSERVATIVE) {
-    PRECICE_DEBUG("Map conservative");
-    static int      mappingIndex = 0;
+    mapConservative(inputDataID, outputDataID, polyparams);
+  } else if (getConstraint() == CONSISTENT) {
+    mapConsistent(inputDataID, outputDataID, polyparams);
+  }
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::mapConservative(int inputDataID, int outputDataID, int polyparams)
+{
+
+  PRECICE_TRACE(inputDataID, outputDataID, polyparams);
+
+  // Gather input data
+  if (utils::MasterSlave::isSlave()) {
+
+    // For conservative, data is not filtered
+    const auto &localInData  = input()->data(inputDataID)->values();
+    const auto &localOutData = output()->data(outputDataID)->values();
+
+    // Send data
+    utils::MasterSlave::_communication->send(localInData.data(), localInData.size(), 0);
+    utils::MasterSlave::_communication->send(localOutData.data(), localOutData.size(), 0);
```

Comment:
Why is `localOutData` also communicated here? That should not be needed, right?
Also: a conservative mapping in parallel is always a write mapping. It goes, thus, from a "provided" to a "received" mesh. Meaning, input is non-overlapping, **but output is**.